src/Swift.Bindings/src/Emitter/StringEmitter/Handler/PropertyHandler.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CodeDom.Compiler;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Factory class for creating instances of PropertyHandler.
+/// </summary>
+public class PropertyHandlerFactory : IFactory<BaseDecl, IPropertyHandler>
+{
+    public bool Handles(BaseDecl decl)
+    {
+        return decl is PropertyDecl;
+    }
+
+    public IPropertyHandler Construct()
+    {
+        return new PropertyHandler();
+    }
+}
+
+/// <summary>
+/// Handler class for property declarations that generates the binding code for Swift properties.
+/// </summary> 
+public class PropertyHandler : BaseHandler, IPropertyHandler
+{
+    private readonly MethodHandler _methodHandler = new();
```

Comment:
Should method handler be created in the factory and then selected based on Handles condition?